contrib/upgrade.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+set -e
```

Comment:
```suggestion
set -euo pipefail
```

http://redsymbol.net/articles/unofficial-bash-strict-mode/

Reply:
good point, done!